From 1b8416d38a544660b52f4b8aa4e0209bcb6a25f9 Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Mon, 24 Jan 2022 09:56:45 -0500
Subject: [PATCH 15/86] ima: Check IMA_MEASURED flag using flags from
 iint_flags()

Use iint_flags() to get the iint flags when checking whether IMA_MEASURED
is set. This change prepares the flag for moving it into the ns_status
structure.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/ima/ima_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/integrity/ima/ima_main.c b/security/integrity/ima/ima_main.c
index 4e3536ec210b..b65a9290e45c 100644
--- a/security/integrity/ima/ima_main.c
+++ b/security/integrity/ima/ima_main.c
@@ -369,7 +369,7 @@ static int __process_measurement(struct ima_namespace *ns,
 			rc = ima_read_modsig(func, buf, size, &modsig);
 
 			if (!rc && ima_template_has_modsig(template_desc) &&
-			    iint->flags & IMA_MEASURED)
+			    iint_flags(iint, ns_status) & IMA_MEASURED)
 				action |= IMA_MEASURE;
 		}
 	}
-- 
2.37.3

